source/deployment/sso-saml-okta.md

```diff
@@ -0,0 +1,124 @@
+
+![okta_10_mattermost_basics.PNG](../../source/images/okta_10_mattermost_basics.PNG)
+
+14) (Optional) Configure Mattermost to verify the signature. The **Service Provider Login URL** is the `Single sign on URL` you specified in Okta in step 7.
+
+![okta_11_mattermost_verification.PNG](../../source/images/okta_11_mattermost_verification.PNG)
+
+15) (Optional) Enable encryption based on the parameters provided in step 8.
+
+![okta_12_mattermost_encryption.PNG](../../source/images/okta_12_mattermost_encryption.PNG)
+
+16) Set attributes for the SAML Assertions, which will be used to update user information in Mattermost. Attributes for email, username, first name and last name are required and should match the values you entered in Okta in step 9. See [documentation on SAML configuration settings](http://docs.mattermost.com/administration/config-settings.html#saml-enterprise) for more detail.
+
+![okta_13_mattermost_attributes.PNG](../../source/images/okta_13_mattermost_attributes.PNG)
+
+17) (Optional) Lastly, customize the login button text.
+
+![okta_14_mattermost_login_button.PNG](../../source/images/okta_14_mattermost_login_button.PNG)
+
+18) Click **Save**.
+
+You’re done! If you’d like to confirm SAML SSO is successfully enabled, switch your System Administrator account from email to SAML-based authentication via **Account Settings > General > Sign-in Method > Switch to SAML SSO** and sign in with your SAML credentials to complete the switch.
+
+It is also recommended to post an announcement about how the migration will work to users.
+
+You may also configure SAML for Okta by editing `config.json`. Before starting the Mattermost server, edit `config.json` to enable SAML based on [SAML configuration settings](http://docs.mattermost.com/administration/config-settings.html#saml-enterprise). You must restart Mattermost server for the changes to take effect.
+
+#### Troubleshooting
+
+The following are troubleshooting suggestions on common error messages and issues. 
+
+##### 1. System Administrator locks themselves out of the system
+
+If the System Administrator is locked out of the system during SAML configuration process, they can set an existing account to System Administrator using [a commandline tool](http://docs.mattermost.com/deployment/on-boarding.html#creating-system-administrator-account-from-commandline). 
+
+##### 2. Received error message: `An account with that username already exists. Please contact your Administrator.`
+
+This usually means an existing account has another authentication method enabled. If so, the user should sign in using that method (such as email and password), then change their sign-in method to SAML via **Account Settings > Security > Sign-in method**.
+
+This error message can also be received if the `Username Attribute` of their SAML credentials is incorrect. If so, the user can update the attribute at their identity provider (for instance, back to the old value if it had been previously updated). 
```

Comment:
I'll change it to ``if the `Username Attribute` of their SAML credentials doesn't match the username of their Mattermost account``